### ISSUE TITLE
It should be possible to use nested lists without any new blank line before

### ISSUE THREAD

**Issue by Vrakfall** [carrot/markdown-styleguide#6]
I think we should be able to use nested lists without inserting a new blank line in between as shown by the following example:
- Something level 1
  - Something level 2
  - Something else level 2
- Something else level 1
- Final level 1 item

``` markdown
- Something level 1
  - Something level 2
  - Something else level 2
- Something else level 1
- Final level 1 item
```

Currently, this style guide forces the insertion of a new line before each level of the list (at least while using the [tidy-markdown](https://atom.io/packages/tidy-markdown) atom package and generates this which, obviously, doesn't have the same result:
- Something level 1
  - Something level 2
  - Something else level 2
- Something else level 1
- Final level 1 item

``` markdown
- Something level 1

  - Something level 2
  - Something else level 2

- Something else level 1
- Final level 1 item
```

**Comment by James-Firth** [comment]
I agree, I'm not a fan of how this functions either. I too am using the atom package in conjunction with the default markdown preview.

(Using code so github doesn't convert the markdown)
I find that the current output of the following is ugly. I find it ugly because it implies a connection between `Sub-item L2` and `Another Item L1` which (in my cases) there is not. 
```
* Item L1

  * Sub-item L2
* Another Item L1

  * Another Item's sub item L2
```

And would personally find either of the following more pleasing. This is because I think `Sub-item L2` should be connected visually to `Item L1` instead.

Compressed
```
* Item L1
  * Sub-item L2
* Another Item L1
  * Another Item's sub item L2
```

Space before next top-level item
```
* Item L1
  * Sub-item L2

* Another Item L1
  * Another Item's sub item L2
```

**Comment by Vrakfall** [comment]
The last one seems even more readable to me, nice idea!

Sorry for the delay, even ents don't speak that slowly. :P
